crates/ggml/src/context.rs

```diff
@@ -70,15 +75,31 @@ impl Context {
             ptr: Arc::new(NonNull::new(raw).expect("Should not be null")),
             mmap: None,
             buffer: None,
+            can_offload: false,
         }
     }
 
+    /// If offloading is enabled, all tensors created by this context will be offloaded to the GPU
+    pub fn enable_offloading(&mut self) {
```

Comment:
Hm, if the only thing that's being changed here is the boolean, I'd suggest changing this to `set_offloading_enabled` or similar. The separate methods make me think that there's additional work being done by each function.

Reply:
Can we change this to `set_offloading(&mut self, can_offload: bool)`? The we only need one function and i would like to keep the boolean field private 🤔